src/import_fieldinformation.py

```diff
@@ -46,122 +47,108 @@ def main(path):
-                          'Importance (1: essential; 2: desirable / recommended; 3: optional)':'importance',
-                          'Comments':'comments',
-                          'Ontologies / references considered':'ontology_reference',
-                          'Link to ontology / reference':'ontology_reference',
-                          'Additional Notes (for development)':'additional_notes',
-                          }
+    column_definitions = {
+        'table':'table',
+        'field':'field',
+        'alias':'alias',
+        'queryset':'queryset',
+        'show in detail':('show_in_detail',True,False,util.bool_converter),
+        'show in list':('show_in_list',True,False,util.bool_converter),
+        'show_as_extra_field':('show_as_extra_field',False,False,util.bool_converter),
+        'is_lincs_field':('is_lincs_field',True,False,util.bool_converter),
+        'is_unrestricted':('is_unrestricted',False,False,util.bool_converter),
+        'list_order':('list_order',True,None,lambda x:util.convertdata(x,int)),
+        'detail_order':('detail_order',True,None,lambda x:util.convertdata(x,int)),
+        'use_for_search_index':('use_for_search_index',True,False,util.bool_converter),
+        'Data Working Group version':'dwg_version',
+        'Unique ID':('unique_id',True),
+        'DWG Field Name':'dwg_field_name',
+        'HMS Field Name':'hms_field_name',
+        'Related to':'related_to',
+        'Description':'description',
+        'Importance (1: essential; 2: desirable / recommended; 3: optional)':'importance',
+        'Comments':'comments',
+        'Ontologies / references considered':'ontology_reference',
+        'Link to ontology / reference':'ontology_reference',
+        'Additional Notes (for development)':'additional_notes',
+        }
        
-    # convert the labels to fleshed out dict's, with strategies for optional, default and converter
-    column_definitions = util.fill_in_column_definitions(properties,column_definitions)
-    
-    num_rows = worksheet.nrows - 1
-    num_cells = worksheet.ncols - 1
+    column_definitions = util.fill_in_column_definitions(
+        properties,column_definitions)
+
+    with open(path) as f:
+        reader = csv.reader(f)
 
-    curr_row = 0 # note zero indexed
-    row = worksheet.row(curr_row)
-    labels = []
-    i = -1
-    while i < num_cells:
-        i += 1
-        # Cell Types: 0=Empty, 1=Text, 2=Number, 3=Date, 4=Boolean, 5=Error, 6=Blank
-        # cell_type = worksheet.cell_type(curr_row, curr_cell)
-        labels.append(str(worksheet.cell_value(curr_row, i)))
+        labels = reader.next()
+        cols = util.find_columns(
+            column_definitions, labels, all_sheet_columns_required=False)
+        
+        logger.info('delete current table');
+        FieldInformation.objects.all().delete()
+        
+        for j,row in enumerate(reader):
+            
+            if(logger.isEnabledFor(logging.DEBUG)): 
```

Comment:
I assume you test the logging level because you're concerned the string manipulations to build the log message could be a performance hit, even when not in debug mode? Remember that the logger interface accepts a %-style format strings and the args to populate it. %s args are stringified automatically, so you can avoid doing expensive string ops unnecessarily just by using that approach. E.g.:

`logger.debug("row %d: %s", j, row)`

That way the string ops get deferred until after the logging code's own check of the current log level, so you don't need your own if-statement. Also I think the code becomes more readable without all the extra sets of parentheses.

Don't bother overhauling all of the logging statements introduced in this PR, but I'd at least clean up this one since the conditional does hurt readability.